src/sympc/protocol/aby3/aby3.py

```diff
@@ -204,3 +207,201 @@ def _get_truncation_pair(x: MPCTensor, session: Session) -> Tuple[MPCTensor]:
         r_mpc = MPCTensor(shares=r, session=session, shape=x.shape)
         rPrime_mpc = MPCTensor(shares=rPrime, session=session, shape=x.shape)
         return r_mpc, rPrime_mpc
+
+    @staticmethod
+    def local_decomposition(
+        x: ReplicatedSharedTensor, ring_size: str, bitwise: bool = False
+    ) -> List[List[List[ReplicatedSharedTensor]]]:
+        """Performs local decomposition to generate shares of shares.
+
+        Args:
+            x (ReplicatedSharedTensor) : input RSTensor.
+            ring_size (str) : Ring size to generate decomposed shares in.
+            bitwise (bool): Perfom bit level decomposition on bits if set.
+
+        Returns:
+            List[ReplicatedSharedTensor]: Decomposed shares in the given ring size.
+
+        Raises:
+            ValueError: If RSTensor does not have session uuid.
+        """
+        if x.session_uuid is None:
+            raise ValueError("Input RSTensor should have session_uuid")
+
+        session = get_session(x.session_uuid)
+        ring_size = int(ring_size)
+        ring_bits = get_nr_bits(session.ring_size)  # for bitwise decomposition
+        tensor_type = get_type_from_ring(ring_size)
+        rank = session.rank
+        nr_parties = session.nr_parties
+
+        zero = torch.zeros(x.shares[0].shape).type(tensor_type)
+
+        # Similar to triples, we have instaces for the shares generated.
+        share_lst: List[List[List[ReplicatedSharedTensor]]] = []
+
+        input_rst = []
+        if bitwise:
+            for i in range(ring_bits):
+                input_rst.append(x.bit_extraction(i))
+        else:
+            input_rst.append(x)
+
+        for share in input_rst:
+            shares = [[zero.clone(), zero.clone()] for i in range(nr_parties)]
+
+            shares[rank][0] = share.shares[0].clone().type(tensor_type)
+
+            shares[(rank + 1) % nr_parties][1] = (
+                share.shares[1].clone().type(tensor_type)
+            )
+
+            rst_sh = []
+            for i in range(nr_parties):
+                rst = x.clone()
+                rst.shares = shares[i]
+                rst.ring_size = ring_size
+                rst_sh.append(rst)
+
+            share_lst.append(rst_sh)
+
+        return share_lst
+
+    @staticmethod
+    def bit_injection(x: MPCTensor, session: Session, ring_size: int) -> MPCTensor:
+        """Perfom ABY3 bit injection for conversion of binary share to arithmetic share.
+
+        Args:
+            x (MPCTensor) : MPCTensor with shares of bit.
+            session (Session): session the share belongs to.
+            ring_size (int) : Ring size of arithmetic share to convert.
+
+        Returns:
+            arith_share (MPCTensor): Arithmetic shares of bit in input ring size.
+
+        Raises:
+            ValueError: If input tensor is not binary shared.
+        """
+        inp_ring = int(x.share_ptrs[0].get_ring_size().get_copy())  # input ring_size
```

Comment:
NIT: Could you name it `input_ring`?

Reply:
It was a change from bit injection PR.I will change it :+1:

Reply:
Done